storage_test.go

```diff
@@ -0,0 +1,442 @@
+			switch aerr.Code() {
+			case dynamodb.ErrCodeResourceInUseException:
+				return aerr
+			case dynamodb.ErrCodeLimitExceededException:
+				return aerr
+			case dynamodb.ErrCodeInternalServerError:
+				return aerr
+			default:
+				return aerr
+			}
+		} else {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func TestDynamoDBStorage_initConfg(t *testing.T) {
+	defaultAwsSession, err := session.NewSession(&aws.Config{
+		Endpoint:   aws.String(""),
+		Region:     aws.String(""),
+		DisableSSL: aws.Bool(DisableSSL),
+	})
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	type fields struct {
+		Table         string
+		KeyPrefix     string
+		ColumnName    string
+		AwsSession    *session.Session
+		AwsEndpoint   string
+		AwsRegion     string
+		AwsDisableSSL bool
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		wantErr  bool
+		expected *Storage
+	}{
+		{
+			name:     "defaults - should error with empty table",
+			fields:   fields{},
+			wantErr:  true,
+			expected: &Storage{},
+		},
+		{
+			name: "defaults - provide only table name",
+			fields: fields{
+				Table: "Testing123",
+			},
+			wantErr: false,
+			expected: &Storage{
+				Table:               "Testing123",
+				AwsSession:          defaultAwsSession,
+				LockTimeout:         lockTimeoutMinutes,
+				LockPollingInterval: lockPollingInterval,
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Storage{
+				Table:         tt.fields.Table,
+				AwsSession:    tt.fields.AwsSession,
+				AwsEndpoint:   tt.fields.AwsEndpoint,
+				AwsRegion:     tt.fields.AwsRegion,
+				AwsDisableSSL: tt.fields.AwsDisableSSL,
+			}
+			if err := s.initConfig(); (err != nil) != tt.wantErr {
+				t.Errorf("initConfig() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			// unset AwsSession since it is too complicated for reflection testing
+			s.AwsSession = tt.expected.AwsSession
+			if !reflect.DeepEqual(tt.expected, s) {
+				t.Errorf("Expected does not match actual: %+v != %+v. \nAwsSession \n\texpected: %+v, \n\tactual: %+v",
+					tt.expected, s, tt.expected.AwsSession, s.AwsSession)
+			}
+		})
+	}
+}
+
+func TestDynamoDBStorage_Store(t *testing.T) {
+	err := initDb()
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	type fields struct {
+		Table         string
+		KeyPrefix     string
+		ColumnName    string
+		AwsSession    *session.Session
+		AwsEndpoint   string
+		AwsRegion     string
+		AwsDisableSSL bool
+	}
+	type args struct {
+		key   string
+		value []byte
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "simple key/value store",
+			fields: fields{
+				Table:         TestTableName,
+				AwsEndpoint:   os.Getenv("AWS_ENDPOINT"),
+				AwsRegion:     os.Getenv("AWS_DEFAULT_REGION"),
+				AwsDisableSSL: DisableSSL,
+			},
+			args: args{
+				key:   "simple-key",
+				value: []byte("value"),
+			},
+			wantErr: false,
```

Comment:
Maybe add a test where wantErr is true?  (e.g. for a blank key?)

Reply:
ok